### PR TITLE
Add DeBERTa

### PR DESCRIPTION
**Pull request by jeswan**
The PR adds DeBERTa V2 support to `jiant`. The PR also includes documentation for how to add a model with the latest changes in `js/feature/easy_add_model`.

### CODE REVIEW

File: jiant/proj/main/modeling/primary.py [hunk 1]
```diff
@@ -291,6 +291,60 @@ def get_feat_spec(self, max_seq_length):
...
+
+    def get_mlm_weights_dict(self, weights_dict):
+        mlm_weights_map = {
+            "lm_predictions.lm_head.bias": "cls.predictions.bias",
+            "lm_predictions.lm_head.dense.weight": "cls.predictions.transform.dense.weight",
+            "lm_predictions.lm_head.dense.bias": "cls.predictions.transform.dense.bias",
+            "lm_predictions.lm_head.LayerNorm.weight": "cls.predictions.transform.LayerNorm.weight",
+            "lm_predictions.lm_head.LayerNorm.bias": "cls.predictions.transform.LayerNorm.bias",
+        }
+        mlm_weights_dict = {new_k: weights_dict[old_k] for new_k, old_k in mlm_weights_map.items()}
```
Comment (zphang) [inline jiant/proj/main/modeling/primary.py:330]: Does `lm_predictions` come from somewhere? The dict here should look like BERT's.

**Review by zphang — CHANGES_REQUESTED** [review]

File: jiant/proj/main/modeling/primary.py (hunk 1 above)
Comment (jeswan) [reply jiant/proj/main/modeling/primary.py]: Looking at this again, I'm not sure where I came up with these keys. I've committed a corrected version.

File: jiant/proj/main/modeling/model_setup.py [hunk 2]
```diff
@@ -160,6 +160,9 @@ def load_encoder_from_transformers_weights(
     for k, v in weights_dict.items():
         if k.startswith(encoder_prefix):
             load_weights_dict[strings.remove_prefix(k, encoder_prefix)] = v
+        elif k.startswith(encoder_prefix.split("-")[0]):
+            # workaround for deberta-v2
```
Comment (zphang) [inline jiant/proj/main/modeling/model_setup.py:164]: Can you add more detail for this comment?

File: guides/models/adding_models.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+ # Adding a model
+
+`jiant` supports or can easily be exteneded to support Hugging Face Hugging Face's [Transformer models](https://huggingface.co/transformers/viewer/) since `jiant` utilizes [Auto Classes](https://huggingface.co/transformers/model_doc/auto.html) to determine the architecture of the model used based on the name of the [pretrained model](https://huggingface.co/models). To add a model not currently supported in `jiant`, follow the following steps:
```
Comment (zphang) [inline guides/models/adding_models.md:3]: Typos: "exteneded", "Hugging Face Hugging Face's"

Maybe add a clarifying explanation (and let me know if this is wrong): We do use AutoModels to resolve the model in jiant, but in order to ensure the jiant pipeline works correctly (e.g. matching the correct tokenizer) and to deal with some subtle differences between models, jiant needs to know some additional information/supports specific handling for specific models, so some additional steps are needed to set up a new model from HF in jiant.

File: guides/models/adding_models.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## (Optional) 4. Add/Register additional task heads
+Specific task heads may require model-specific implementation such as the MLM task heads. These model-specific task heads should be added and registered with their respective factory in (jiant/proj/main/modeling/heads.py)[../../jiant/proj/main/modeling/heads.py] if applicable. For example, MLM heads require a factory since their implementation differs across models:
+
+```python
+@JiantMLMHeadFactory.register([ModelArchitectures.DEBERTAV2])
+class DebertaV2MLMHead(BaseMLMHead):
+    ...
+````
+
```
Comment (zphang) [inline guides/models/adding_models.md:70]: Conclude with "you should now be able to ..."

**Review by zphang — APPROVED** [review]
Minor documentation comments.

**Pull request merged**